settings_display: Add spacing to the `Save` and `Add Emoji` button

ankur12-1610 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Fixes: #20118 

**Testing plan:** <!-- How have you tested? -->
Manually

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
#### For Save button-
Before:
![Screenshot-20211103123908-597x179](https://user-images.githubusercontent.com/76884959/140024932-7f89d75e-a510-4d96-86ff-37dfe4bb26a6.png)

After:
![Screenshot-20211103123838-595x193](https://user-images.githubusercontent.com/76884959/140024951-e07ee777-fdb9-4a9b-9ba8-1122691c80e4.png)

#### For Add Emoji button-
Before:
![Screenshot-20211103162949-599x189](https://user-images.githubusercontent.com/76884959/140050633-42203b14-a52b-444f-94a9-6e7af79f8bc2.png)
<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

After:
![Screenshot-20211103163016-595x192](https://user-images.githubusercontent.com/76884959/140050707-087f12ff-b1d6-4f7e-8080-49b9730c726a.png)

amanagr commented on static/styles/app_components.css:
```diff
@@ -65,7 +65,7 @@ input::placeholder {
     .button {
         padding: 7px 14px;
-        margin: 0;
+        margin: 0.25rem 0;
```
You shouldn't modify a general class like `.button` for a specific button. Also, you should use `px` unit as that is what we are using right now. There should be a full stop in your commit message.

ankur12-1610 replied:
Ah! got it.

@amanagr I added a new class named button-spacer which solved the issue.

Also, I found another button in Custom emoji section with same issue so I'm adding it in this PR itself.

Kindly review the changes :)

amanagr commented on static/styles/app_components.css:
```diff
@@ -98,6 +98,10 @@ input::placeholder {
             border-radius: 4px;
         }
 
+        &.button-spacer {
```
Instead of adding a new class, you can reference the button as like `.new-emoji-form #admin_emoji_submit`, also the full stop in commit message.

ankur12-1610 replied:
> Instead of adding a new class, you can reference the button as like `.new-emoji-form #admin_emoji_submit`, also the full stop in commit message.

Done! :) 

Can you verify if I have done it correctly?

amanagr commented:
<img width="858" alt="Screenshot 2021-11-06 at 8 40 27 AM" src="https://user-images.githubusercontent.com/25124304/140596061-09775e25-8f8f-4c01-9b3b-dc4d1e260465.png">

You need to use media query here, you are breaking the alignment on wide screens.

ankur12-1610 commented:
> <img alt="Screenshot 2021-11-06 at 8 40 27 AM" width="858" src="https://user-images.githubusercontent.com/25124304/140596061-09775e25-8f8f-4c01-9b3b-dc4d1e260465.png">
> 
> You need to use media query here, you are breaking the alignment on wide screens.

@amanagr I have fixed the following. I added an id to some of the divs because I was facing issues in some media queries and by introducing these ids it became easier.

Could you please verify? :)

amanagr commented:
<img width="513" alt="Screenshot 2021-11-09 at 8 30 21 AM" src="https://user-images.githubusercontent.com/25124304/140854045-1c4e925e-294f-4f41-a941-fef386af16e2.png">

Can you also fix this? I still think using media queries would be required.

ankur12-1610 commented:
> <img alt="Screenshot 2021-11-09 at 8 30 21 AM" width="513" src="https://user-images.githubusercontent.com/25124304/140854045-1c4e925e-294f-4f41-a941-fef386af16e2.png">
> 
> Can you also fix this? I still think using media queries would be required.

@amanagr I have used media queries :sweat_smile: I haven't used made changes globally.

And regarding the `preview text` I have solved it by adding more spacing to the bottom.

And I have also altered the right alignment of `Upload Emoji` and `Clear Emoji` button as it was misaligned in few media queries.

When I only added margin to buttons I faced problems in iPad media queries.

So I altered the margins of other elements.

amanagr commented:
@ankur12-1610 Thanks! Looks good to me overall. I think Tim would prefer `class=user_name` instead of `id="user_name_div` (similarly for others) and then use `.new-user-group-form div.user_name` etc.. before merging this. So, it will be good to do that change.

ankur12-1610 commented:
> @ankur12-1610 Thanks! Looks good to me overall. I think Tim would prefer `class=user_name` instead of `id=user_name_div` (similarly for others) and then use `.new-user-group-form div.user_name` etc.. before merging this. So, it will be good to do that change.

@amanagr done! :)
I have given `class` and `id` names the same value, is it ok?
Kindly look into it!

timabbott commented:
This seems like a lot of hand-written CSS for specific elements; is there a way we could do something more generic by making a single set of shared CSS for buttons in these form components?

ankur12-1610 commented:
> This seems like a lot of hand-written CSS for specific elements; is there a way we could do something more generic by making a single set of shared CSS for buttons in these form components?



> This seems like a lot of hand-written CSS for specific elements; is there a way we could do something more generic by making a single set of shared CSS for buttons in these form components?

@timabbott I did try to minimize the manual CSS, but the alignment of some elements gets weird and it requires individual spacing. I tried to group the elements as much as I could  :sweat_smile:
